### PR TITLE
Added finnish locale

### PR DESCRIPTION
**Pull request by lordpansar**
[lordpansar](https://github.com/lordpansar) Medium [lordpansar /master → Lissy93/dashy](https://github.com/Lissy93/dashy/tree/master) Commits: 2 | Files Changed: 2 | Additions: 285 Label Unchecked Tasks [Powered by Pull Request Badge](https://pullrequestbadge.com/?utm_medium=github&utm_source=Lissy93&utm_campaign=badge_info)

*Thank you for contributing to Dashy! So that your PR can be handled effectively, please populate the following fields (delete sections that are not applicable)*

**Category**: 
> One of: Other (please specify) - Added finnish locale

**Overview**
> Added finnish language locale

**Issue Number** _(if applicable)_ #00

**New Vars** _(if applicable)_
> N/A

**Screenshot** _(if applicable)_
> N/A

**Code Quality Checklist** _(Please complete)_
- [x] All changes are backwards compatible
- [x] All lint checks and tests are passing
- [x] There are no (new) build warnings or errors
- [ ] _(If a new config option is added)_ Attribute is outlined in the schema and documented
- [ ] _(If a new dependency is added)_ Package is essential, and has been checked out for security or performance
- [ ] Bumps version, if new feature added

### CODE REVIEW

**Comment by lordpansar** [comment]
Bump :)

**Review by Lissy93 — COMMENTED** [review]
Hiya @lordpansar - This looks awesome, thanks for adding! And sorry for the late review/ 

There's just one small thing missing - add the following into [`/src/utils/languages.js`](https://github.com/lordpansar/dashy/blob/master/src/utils/languages.js)


```javascript
import fi from '@/assets/locales/fi.json';
```

and

```javascript
  { // Finish
    name: 'Suomalainen',
    code: 'fi',
    locale: fi,
    flag: '🇫🇮',
  },
  ```

There's a guide on how to do this [here](https://github.com/Lissy93/dashy/blob/master/docs/multi-language-support.md#adding-a-new-language), but let me know if you still need any help

File: src/assets/locales/fi.json [hunk 1]
```diff
@@ -0,0 +1,284 @@
...
+    "backup-id-label":"Varatunnuksesi",
+    "backup-id-note":"Tätä käytetään palauttamiseen varmuuskopioista myöhemmin. Pidä se siis yhdessä salasanasi kanssa turvallisessa paikassa.",
+    "restore-title":"Palauta varmuuskopio",
+    "restore-id-label":"Palauta ID",
+    "restore-password-label":"Salasana",
+    "restore-button":"Nollaa",
+    "backup-missing-password":"Salasana puuttuu",
+    "backup-error-unknown":"Pyyntöä ei voida käsitellä",
+    "backup-error-password":"Väärä salasana. Anna nykyinen salasanasi.",
+    "backup-success-msg":"Valmistettu ilman ongelmia",
```
Comment (yesBad) [inline src/assets/locales/fi.json:167]: "Toteuttettu onnistuneesti"

File: src/assets/locales/fi.json [hunk 2]
```diff
@@ -0,0 +1,284 @@
...
+    "restore-success-msg":"Kokoonpano on palautettu ilman ongelmia"
```
Comment (yesBad) [inline src/assets/locales/fi.json:168]: "Asetukset palautettu onnistuneesti"

File: src/assets/locales/fi.json [hunk 3]
```diff
@@ -0,0 +1,284 @@
...
+  },
+  "menu":{
+    "open-section-title":"Avaa",
+    "sametab":"Tämä välilehti",
+    "newtab":"Uusi välilehti",
+    "modal":"Pop-Up Modal",
+    "workspace":"Workspace-näkymä",
```
Comment (yesBad) [inline src/assets/locales/fi.json:175]: "Työpöytä-näkymä"

File: src/assets/locales/fi.json [hunk 4]
```diff
@@ -0,0 +1,284 @@
...
+    "backup-error-unknown":"Pyyntöä ei voida käsitellä",
+    "backup-error-password":"Väärä salasana. Anna nykyinen salasanasi.",
+    "backup-success-msg":"Valmistettu ilman ongelmia",
+    "restore-success-msg":"Kokoonpano on palautettu ilman ongelmia"
+  },
+  "menu":{
+    "open-section-title":"Avaa",
+    "sametab":"Tämä välilehti",
+    "newtab":"Uusi välilehti",
+    "modal":"Pop-Up Modal",
```
Comment (yesBad) [inline src/assets/locales/fi.json:174]: "Ponnahdusikkuna modaali", but yeah it sounds kinda better in English

File: src/assets/locales/fi.json [hunk 5]
```diff
@@ -0,0 +1,284 @@
...
+    "workspace":"Workspace-näkymä",
+    "options-section-title":"Vaihtoehtoinen",
+    "edit-item":"Muokata",
```
Comment (yesBad) [inline src/assets/locales/fi.json:177]: Muokkaa

File: src/assets/locales/fi.json [hunk 6]
```diff
@@ -0,0 +1,284 @@
...
+    "move-item":"Kopioi tai siirrä",
+    "remove-item":"Poista"
+  },
+  "context-menus":{
+    "item":{
+      "open-section-title":"Avaa",
+      "sametab":"Tämä välilehti",
+      "newtab":"Uusi välilehti",
+      "modal":"Pop-Up Modal",
```
Comment (yesBad) [inline src/assets/locales/fi.json:186]: "Ponnahdusikkuna modaali", but yeah it sounds kinda better in English

File: src/assets/locales/fi.json [hunk 7]
```diff
@@ -0,0 +1,284 @@
...
+      "workspace":"Workspace View",
```
Comment (yesBad) [inline src/assets/locales/fi.json:187]: "Työpöytä-näkymä"

File: src/assets/locales/fi.json [hunk 8]
```diff
@@ -0,0 +1,284 @@
...
+      "options-section-title":"Vaihtoehtoinen",
+      "edit-item":"Muokata",
```
Comment (yesBad) [inline src/assets/locales/fi.json:189]: Muokkaa

File: src/assets/locales/fi.json [hunk 9]
```diff
@@ -0,0 +1,284 @@
...
+      "move-item":"Kopioi tai siirrä",
+      "remove-item":"Poista"
+    },
+    "section":{
+      "open-section":"Avaa osio",
+      "edit-section":"Muokata",
```
Comment (yesBad) [inline src/assets/locales/fi.json:195]: Muokkaa

File: src/assets/locales/fi.json [hunk 10]
```diff
@@ -0,0 +1,284 @@
...
+      "move-section":"Muuttaa",
```
Comment (yesBad) [inline src/assets/locales/fi.json:196]: Siirtää

File: src/assets/locales/fi.json [hunk 11]
```diff
@@ -0,0 +1,284 @@
...
+    }
+  },
+  "interactive-editor":{
+    "menu":{
+      "start-editing-tooltip":"Avaa interaktiivinen editori",
+      "edit-site-data-subheading":"Muokkaa sivuston tietoja",
+      "edit-page-info-btn":"Muokkaa sivun tietoja",
+      "edit-page-info-tooltip":"Sovelluksen nimi, kuvaus, navigointilinkit, alatunnisteen teksti jne",
+      "edit-app-config-btn":"Muokkaa sovelluksen asetuksia",
+      "edit-app-config-tooltip":"Muut sovelluksen määritysvaihtoehdot",
```
Comment (yesBad) [inline src/assets/locales/fi.json:207]: "Muut sovelluksen asetukset"

File: src/assets/locales/fi.json [hunk 12]
```diff
@@ -0,0 +1,284 @@
...
+      "config-save-methods-subheading":"Kokoonpanon tallennusvaihtoehto",
+      "save-locally-btn":"Tallenna paikallisesti",
+      "save-locally-tooltip":"Tallenna asetukset paikallisesti selaimen tallennustilaan. Tämä ei vaikuta asetustiedostoosi, mutta muutokset tallennetaan vain tälle laitteelle",
+      "save-disk-btn":"Tallenna levylle",
+      "save-disk-tooltip":"Tallenna kokoonpano levyllä olevaan conf.yml-tiedostoon. Tämä varmuuskopioi ja korvaa nykyiset määritykset",
```
Comment (yesBad) [inline src/assets/locales/fi.json:212]: "määritykset" -> "asetukset"

File: src/assets/locales/fi.json [hunk 13]
```diff
@@ -0,0 +1,284 @@
...
+      "export-config-btn":"Vie kokoonpano",
```
Comment (yesBad) [inline src/assets/locales/fi.json:213]: "kokoonpano" -> "asetukset"

File: src/assets/locales/fi.json [hunk 14]
```diff
@@ -0,0 +1,284 @@
...
+      "export-config-tooltip":"Tarkastele ja vie uutta kokoonpanoa joko tiedostoon tai leikepöydälle",
```
Comment (yesBad) [inline src/assets/locales/fi.json:214]: "Tarkastele ja vie uutta kokoonpanoa" -> "Tarkastele ja vie uudet asetukset"

File: src/assets/locales/fi.json [hunk 15]
```diff
@@ -0,0 +1,284 @@
...
+      "cloud-backup-btn":"Varmuuskopio pilveen",
+      "cloud-backup-tooltip":"Tallenna määritysten salattu varmuuskopio pilveen",
```
Comment (yesBad) [inline src/assets/locales/fi.json:216]: "määritysten" -> "asetuksien"

File: src/assets/locales/fi.json [hunk 16]
```diff
@@ -0,0 +1,284 @@
...
+      "edit-raw-config-btn":"Muokkaa raw-määritystä",
```
Comment (yesBad) [inline src/assets/locales/fi.json:217]: "Muokkaa raaka-asetuksia"

File: src/assets/locales/fi.json [hunk 17]
```diff
@@ -0,0 +1,284 @@
...
+      "edit-raw-config-tooltip":"Tarkastele ja muokkaa raw-määritystä JSON-editorin kautta",
+      "cancel-changes-btn":"Peruuta muokkaus",
+      "cancel-changes-tooltip":"Poista nykyiset muutokset ja poistu muokkaustilasta. Tämä ei vaikuta tallennettuihin määrityksiin.",
+      "edit-mode-name":"Muokkaustila",
+      "edit-mode-subtitle":"Olet muokkaustilassa",
+      "edit-mode-description":"Tämä tarkoittaa, että voit tehdä muutoksia asetuksiin ja esikatsella tuloksia, mutta mitään muutoksia ei säilytetä, ennen kuin tallennat.",
+      "save-stage-btn":"Tallentaa",
```
Comment (yesBad) [inline src/assets/locales/fi.json:224]: Tallenna

File: src/assets/locales/fi.json [hunk 18]
```diff
@@ -0,0 +1,284 @@
...
+      "cancel-stage-btn":"Peruuttaa"
```
Comment (yesBad) [inline src/assets/locales/fi.json:225]: Peruuta

File: src/assets/locales/fi.json [hunk 19]
```diff
@@ -0,0 +1,284 @@
...
+      "edit-section-title":"Muokkaa osiota",
+      "add-section-title":"Lisää uusi osio",
+      "edit-tooltip":"Napauta muokataksesi tai napsauta hiiren kakkospainikkeella nähdäksesi lisää vaihtoehtoja",
+      "remove-confirm":"Haluatko varmasti poistaa tämän osion? Tämä toiminto voidaan kumota myöhemmin."
+    },
+    "edit-app-config":{
+      "warning-msg-title":"Etene varovaisesti",
+      "warning-msg-l1":"Seuraavat vaihtoehdot koskevat edistyneitä sovellusmäärityksiä.",
+      "warning-msg-l2":"Jos olet epävarma jostakin kentästä, tarkista",
+      "warning-msg-docs":"dokumentaatiota",
```
Comment (yesBad) [inline src/assets/locales/fi.json:240]: tarkista "dokumentaatiota" doesn't sound natural -> "dokumentaatiosta"

File: src/assets/locales/fi.json [hunk 20]
```diff
@@ -0,0 +1,284 @@
...
+      "warning-msg-l3":"ei-toivottujen seurausten välttämiseksi."
+    },
+    "export":{
+      "export-title":"Vie kokoonpano",
```
Comment (yesBad) [inline src/assets/locales/fi.json:244]: "kokoonpano" -> "asetukset"

File: src/assets/locales/fi.json [hunk 21]
```diff
@@ -0,0 +1,284 @@
...
+      "copy-clipboard-btn":"Kopioi leikepöydälle",
+      "copy-clipboard-tooltip":"Kopioi kaikki sovellusmääritykset järjestelmän leikepöydälle YAML-muodossa",
```
Comment (yesBad) [inline src/assets/locales/fi.json:246]: "sovellusmääritykset" -> "sovellusasetukset"

File: src/assets/locales/fi.json [hunk 22]
```diff
@@ -0,0 +1,284 @@
...
+      "download-file-btn":"Lataa tiedostona",
+      "download-file-tooltip":"Lataa kaikki sovellusmääritykset laitteellesi YAML-tiedostona",
```
Comment (yesBad) [inline src/assets/locales/fi.json:248]: "sovellusmääritykset" -> "sovellusasetukset"

**Review by yesBad — CHANGES_REQUESTED** [review]
I looked over these quiet quickly, but it certainty looks like using some AI/Translate to just get PR points sadly. Using words that no actual Finnish person would use really and has some old fragments of Swedish for whatever reason? I hope it's just a bad Finnish translation.... Goodluck 👍

**Review by yesBad — CHANGES_REQUESTED** [review]
I looked over these quiet quickly, but it certainty looks like using some AI/Translate to just get PR points sadly. Using words that no actual Finnish person would use really and has some old fragments of Swedish for whatever reason? I hope it's just a bad Finnish translation.... Goodluck 👍

**Comment by lordpansar** [comment]
Hi @yesBad !

Thank you for your input and review. Finnish is not my first language although I speak it. I used my head where I could and looked up some words I didn't know online. Apparently this wasn't up to par for a native speaker regarding choice of words etc.

It makes me kind of sad though that you jump to conclusions when I'm trying to contribute to a piece of OSS that I enjoy.

I was also hoping that I could provide a foundation where a native speaker would catch improper grammar/spelling and update them in a PR. Just like I did when I updated the Swedish locale 1-2 years ago.

I'll let a family member with better Finnish proficency go through this. Thanks once again for your review 😊

**Comment by yesBad** [comment]
> Hi @yesBad !
> 
> Thank you for your input and review. Finnish is not my first language although I speak it. I used my head where I could and looked up some words I didn't know online. Apparently this wasn't up to par for a native speaker regarding choice of words etc.
> 
> It makes me kind of sad though that you jump to conclusions when I'm trying to contribute to a piece of OSS that I enjoy.

Sorry about that, just a lot of Finnish translations seem to be done with some AI/Translate API nowdays sadly. 👍 

Goodluck with the PR.

**Comment by lordpansar** [comment]
np bro 😊

**Pull request closed without merging**